ambry-network/src/main/java/com/github/ambry/network/http2/Http2NetworkClient.java

```diff
@@ -75,6 +76,9 @@ public Http2NetworkClient(Http2ClientMetrics http2ClientMetrics, Http2ClientConf
         new StreamChannelInitializer());
     this.http2ClientMetrics = http2ClientMetrics;
     correlationIdInFlightToChannelMap = new ConcurrentHashMap<>();
+    Gauge<Long> nettyPendingTasks = () -> Http2Utils.getNumberOfPendingTasks(eventLoopGroup);
+    http2ClientMetrics.registry.register(MetricRegistry.name(Http2NetworkClient.class, "NettyPendingTasks"),
+        nettyPendingTasks);
```

Comment:
This piece of code can be moved to `Http2ClientMetrics`. You can either initialize `Http2ClientMetrics` within `Http2NetworkClient` (thus you can pass in Http2Client or eventLoopGroup) or  you can create a method something like `initializeEventLoopGroupMetrics(eventLoopGroup)` in `Http2ClientMetrics` and call it here.